### PR TITLE
Cluster certificate path

### LINKED ISSUE #8708: Allow passing a path to a cert on disk when joining a cluster
This is not a bug, it's a feature request.

The cluster bootstrap docs here show how to join a cluster.https://lxd.readthedocs.io/en/latest/clustering/

But it seems that embedding a pem-encoded cert inside yaml is our only option.

Templating yaml with large strings (like certificates) is error prone. It would be friendlier to automation if a filepath could be specified instead. This would avoid the need to run `sed` to post-process the cluster cert, and embed it in yaml.

Perhaps something like "cluster_certificate_path"

```yaml
cluster:
  enabled: true
  server_name: <%= $server_name %>
  server_address: <%= $server_addr %>
  cluster_address: <%= $cluster_addr %>
  cluster_certificate_path: "<%= $certificate_path %>"
  cluster_password: <%= $trust_password %>
```

### PR DESCRIPTION
**Pull request by presztak**
Resolves #8708

### CODE REVIEW

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
I don't think we want to modify `shared/api` here as it wouldn't make much sense for the REST API to take in a file name.

Where we really want this is for `lxd init --preseed` and to handle that, the best would be to put the new field in `initDataCluster` which is a struct defined in `lxd/init.go` and that's currently only made of `api.ClusterPut`.

Adding the field to `initDataCluster` will allow the exact same preseeding that you've allowed with your branch but without adding a field to the external REST API. The new field should be mentioned in `doc/clustering.md` as an alternative to including the certificate directly in the YAML.

File: lxd/main_init.go [hunk 1]
```diff
@@ -128,6 +129,19 @@ func (c *cmdInit) Run(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	// Check if the path to the cluster certificate is set
+	// If yes then read cluster certificate from file
+	if config.Cluster != nil && len(config.Cluster.ClusterCertificatePath) > 0 {
```
Comment (stgraber) [inline lxd/main_init.go:134]: We usually do `config.Cluster.ClusterCertificatePath != ""` instead of looking at the string length.

**Comment by stgraber** [comment]
@presztak thanks, looks good, just one tiny nit about how to check whether the key was set. (`!= ""` instead of `len() > 0`)

**Comment by presztak** [comment]
@stgraber done

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
Thanks!

**Pull request merged**